reco_utils/evaluation/python_evaluation.py

```diff
@@ -531,11 +531,17 @@ def map_at_k(
     threshold=DEFAULT_THRESHOLD,
 ):
     """
+    The implementation of the MAP is referenced from Spark MLlib evaluation metrics.
+    https://spark.apache.org/docs/2.3.0/mllib-evaluation-metrics.html#ranking-systems
+
     Get mean average precision at k. A good reference can be found at
-    https://people.cs.umass.edu/~jpjiang/cs646/03_eval_basics.pdf
+    http://web.stanford.edu/class/cs276/handouts/EvaluationNew-handout-6-per.pdf
 
-    NOTE: The MAP is at k because the evaluation class takes top k items for
-    the prediction items.
+    Note:
+        1. The MAP is at k because the evaluation class takes top k items for
+        the prediction items.
+        2. The MAP is to calculate the AP in the relevant items, so it is normalized
```

Comment:
I would say: The MAP is meant to calculate Avg. Precision for the relevant items, so it is normalized by the number of relevant items in the ground truth data, instead of k.

Reply:
Thanks @gramhagen this is more precise! The original note #1 is to explain the name of the function as in some context (eg Spark) it is called just “MAP” instead of “MAP@k”. I think it also makes sense as it indeed is not normalised by k.